docs_src/security/tutorial004.py

```diff
@@ -119,7 +121,7 @@ async def login_for_access_token(form_data: OAuth2PasswordRequestForm = Depends(
         raise HTTPException(
             status_code=status.HTTP_401_UNAUTHORIZED,
             detail="Incorrect username or password",
-            headers={"WWW-Authenticate": "Bearer"},
+            headers={"WWW-Authenticate": "Basic"},
```

Comment:
This made me think for a bit, but I think I agree with you. Although we are using helper classes with `OAuth2` stuff in them, it is basically a basic authentication. On the other hand, the username/password are following the Oauth2 standards (e.g. with the form) and not with base64 encoding (as is required by the Basic Authentication standard). So, it can be both ways I guess?

Reply:
Hmm I don't know. Somehow it's neither of the two options. I found [this](https://stackoverflow.com/questions/1748374/http-401-whats-an-appropriate-www-authenticate-header-value) on this topic. I guess there is no simple soultion. Maybe just leave the WWW-Authenticate out for simplicities sake or leave as is?
Bearer is definitely false in my opinion.

Reply:
When using basic auth, the browser prompts you with a username/passwd form. This might not even work, as the browser will probably put the credentials into the header instead of the body. I didn't test this out yet